src/zimwriterfs/zimcreatorfs.cpp

```diff
@@ -26,9 +26,32 @@
 #include <dirent.h>
 #include <sys/stat.h>
 #include <regex>
+#include <unistd.h>
+#include <limits.h>
+#include <cassert>
 
 bool isVerbose();
 
+ZimCreatorFS::ZimCreatorFS(std::string _directoryPath, std::string mainPage,
+                           bool verbose, bool uniqueNamespace, bool zstd)
+  : zim::writer::Creator(verbose, zstd ? zim::zimcompZstd : zim::zimcompLzma),
+    init_success(false),
+    directoryPath(_directoryPath),
+    mainPage(mainPage),
+    uniqueNamespace(uniqueNamespace)
+{
+  char buf[PATH_MAX];
+
+  if (realpath(directoryPath.c_str(), buf) != buf) {
+    std::cerr << "Unable to canonicalize HTML directory path " << directoryPath
+              << ": " << strerror(errno) << std::endl;
+    return;
```

Comment:
I try to avoid possibilities of creating invalid objects.
If the arguments of the constructor of `ZimCreatorFS` are not valid, we must not create a `ZimCreatorFS`.

It would be better to raise an exception here.

Reply:
Done, an exception is raised.